appinventor/appengine/src/com/google/appinventor/server/DownloadServlet.java

```diff
@@ -184,9 +186,17 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOExc
           zipName = "u" + projectUserId + "_p" + projectId + ".aia";
         }
         ProjectSourceZip zipFile = fileExporter.exportProjectSourceZip(projectUserId,
-          projectId, /* include history*/ true, /* include keystore */ true, zipName, true, true, false, false);
+                projectId, /* include history*/ true, /* include keystore */ true, zipName, true, true, false, false);
+        downloadableFile = zipFile.getRawFile();
+      } else if (downloadKind.equals(ServerLayout.DOWNLOAD_SELECTED_PROJECTS_SOURCE)) {
+        String[] projectIdStrings = uriComponents[PROJECT_ID_INDEX].split("-");
```

Comment:
@ewpatton I found [this](https://stackoverflow.com/questions/35277580/url-size-in-google-app-engine) StackOverflow question that talks about the maximum URI size of a request In Google App Engine. Is that big enough? If it isn't, we should probably switch to using a POST request, but this will also require changes to how the request is called with GWT.